api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirLocationDaoImpl.java

```diff
@@ -112,8 +119,56 @@ private void handleTag(Criteria criteria, TokenAndListParam tags) {
 		}
 	}
 	
-	private void handleParentLocation(Criteria criteria, ReferenceAndListParam parent) {
-		handleLocationReference("loc", parent).ifPresent(loc -> criteria.createAlias("parentLocation", "loc").add(loc));
+	private void handleLocationReference(Criteria criteria, ReferenceAndListParam locationAndReferences) {
+		
+		if (locationAndReferences == null) {
```

Comment:
All this through 148 is basically a series null / size checks to make sure that ReferenceAnd list only contains a single reference.  We probably should change this parameter to just we a